src/controllers/LocalStorageHistoryController.ts

```diff
@@ -55,26 +56,52 @@ export class LocalStorageHistoryController extends RootComponent implements IHis
     this.omit = attributes;
   }
 
+  public setState(values: Record<string, any>) {
+    this.storage.save(values);
+  }
+
   private updateLocalStorageFromModel() {
-    const attributes = _.omit(this.model.getAttributes(), this.omit);
+    const attributes = omit(this.model.getAttributes(), this.omit);
     this.setState(attributes);
     this.logger.debug('Saving state to localstorage', attributes);
   }
 
-  private updateModelFromLocalStorage() {
-    const toSet: { [key: string]: any } = {};
-    const loadedFromStorage = this.storage.load();
-    _.each(<_.Dictionary<any>>this.model.attributes, (value, key?, obj?) => {
-      var valToSet = loadedFromStorage ? loadedFromStorage[key] : undefined;
-      if (valToSet == undefined) {
-        valToSet = this.model.defaultAttributes[key];
-      }
-      toSet[key] = valToSet;
+  private initModelFromLocalStorageAndUrlHash() {
+    const model = {
+      ...this.localStorageModel,
+      ...this.urlHashModel
+    };
+
+    this.model.setMultiple(model);
+  }
+
+  private get localStorageModel() {
+    const model: Record<string, any> = {};
+    const storedValues = this.storage.load() || {};
+
+    each(this.model.attributes, (value, key) => {
+      const storedValue = storedValues[key];
+      const defaultValue = this.model.defaultAttributes[key];
+      const valueToSet = storedValue == undefined ? defaultValue : storedValue;
+
+      model[key] = valueToSet;
     });
-    this.model.setMultiple(toSet);
+
+    return model;
   }
 
-  public setState(values: Record<string, any>) {
-    this.storage.save(values);
+  private get urlHashModel(): Record<string, any> {
+    const model: Record<string, any> = {};
+    const hash = HashUtils.getHash(this.windoh);
+
+    each(this.model.attributes, (value, key) => {
+      const valueInUrl = HashUtils.getValue(key, hash);
+
+      if (valueInUrl) {
```

Comment:
Is it possible for a hash value to be another falsy value such as an empty string or 0? If so, is there a scenario where we might want those values to take priority over the local storage's? Would it be a good idea to add a test for that scenario?